daemon/daemon.go

```diff
@@ -358,10 +354,12 @@ func (d *Daemon) Start() {
 		case state.RestartDaemon:
 			d.tomb.Kill(nil)
 		case state.RestartSystem:
-			cmd := exec.Command("shutdown", "+10", "-r", shutdownMsg)
-			if out, err := cmd.CombinedOutput(); err != nil {
-				logger.Noticef("%s", osutil.OutputErr(out, err))
+			bootloader, err := partition.FindBootloader()
+			if err != nil {
+				logger.Noticef("cannot get bootloader: %s", err)
+				break
```

Comment:
that's a much bigger issue than this place in the code (there various places assuming that if we are core system we will get a reboot in some situations), if we have such issues we need to think through them, this is probably not the right PR for that